runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -118,17 +122,45 @@ private void translateKafkaInput(
     }
 
     final String topic;
-    Preconditions.checkNotNull(topic = (String) params.get("topic"), "'topic' needs to be set");
+    final String groupId;
+    final String bootstrapServers;
 
+    Preconditions.checkNotNull(topic = (String) params.get("topic"), "'topic' needs to be set");
     Map<?, ?> consumerProps = (Map) params.get("properties");
     Preconditions.checkNotNull(consumerProps, "'properties' need to be set");
-    final Properties properties = new Properties();
-    properties.putAll(consumerProps);
 
-    logger.info("Kafka consumer for topic {} with properties {}", topic, properties);
+    logger.info("Configuring kafka consumer for topic {} with properties {}", topic, consumerProps);
+
+    Preconditions.checkNotNull(
+        groupId = (String) consumerProps.remove("group.id"), "'group.id' needs to be set");
+    Preconditions.checkNotNull(
+        bootstrapServers = (String) consumerProps.remove("bootstrap.servers"),
+        "'bootstrap.servers' needs to be set");
+
+    final String splits[] = bootstrapServers.split(":", 2);
+    final String bootstrapServer = splits[0];
+    int port = splits.length == 2 ? Integer.parseInt(splits[1]) : DEFAULT_KAFKA_BROKER_PORT;
+    final String userName = (String) params.get("username");
+    final String password = (String) params.get("password");
+
+    LyftKafkaConsumerBuilder<WindowedValue<byte[]>> consumerBuilder =
+        new LyftKafkaConsumerBuilder<>();
+    if (userName != null) {
+      consumerBuilder.withUsername(userName);
+    }
+
+    if (password != null) {
+      consumerBuilder.withPassword(password);
+    }
+
+    for (Map.Entry<?, ?> entry : consumerProps.entrySet()) {
+      consumerBuilder.withKafkaProperty(
```

Comment:
Property keys are always of type String.

Reply:
Instead of having this loop, it would be better to change the builder to `withKafkaProperties`